edit.ml

```diff
@@ -44,3 +77,122 @@ let subst_many instrs substs =
+  let split_edge = [|
+    Label split_label;
+    Goto label;
+  |] in
+  (* insert the split edge *)
+  let new_instrs, pc_map = subst instrs pc' 0 split_edge in
+  let new_branch = match[@warning "-4"] instrs.(pc) with
+    | Branch (v, l1, l2) ->
+      if l1 = label
+      then Branch (v, split_label, l2)
+      else if l2 = label
+      then Branch (v, l1, split_label)
+      else invalid_arg "split_edge"
+    | _ -> invalid_arg "split_edge" in
+  new_instrs.(pc_map pc) <- new_branch;
+  new_instrs, pc_map
+
+let replace_uses_in_instruction old_name new_name instr : instruction =
+  let in_simple_expression (exp:simple_expression) : simple_expression =
+    match exp with
+    | Constant _ -> exp
+    | Var x -> if x = old_name then Var new_name else exp in
+  let in_expression exp : expression =
+    match exp with
+    | Simple se -> Simple (in_simple_expression se)
+    | Op (op, exps) ->
+      Op (op, List.map in_simple_expression exps) in
+  let in_arg exp : argument =
+    match exp with
+    | Arg_by_val e -> Arg_by_val (in_expression e)
+    | Arg_by_ref x -> if x = old_name then Arg_by_ref new_name else Arg_by_ref x in
+  let in_osr osr : osr_def =
+    match osr with
+    | Osr_const (x, exp) -> Osr_const (x, in_expression exp)
+    | Osr_mut (x, y) -> if y = old_name then Osr_mut (x, new_name) else osr
+    | Osr_mut_undef _ -> osr in
+
+  match instr with
+  | Call (x, f, exs) ->
+    assert(x != old_name);   (* -> invalid scope *)
+    Call (x, in_expression f, List.map in_arg exs)
+  | Stop e ->
+    Stop (in_expression e)
+  | Return e ->
+    Return (in_expression e)
+  | Decl_const (x, exp) ->
+    assert(x != old_name);   (* -> invalid scope *)
+    Decl_const (x, in_expression exp)
+  | Decl_mut (x, Some exp) ->
+    assert (x != old_name);
+    Decl_mut (x, Some (in_expression exp))
+  | Assign (x, exp) ->
+    Assign (x, in_expression exp)
+  | Drop x ->
+    if x = old_name then Drop new_name else instr
+  | Clear x ->
+    if x = old_name then Clear new_name else instr
+  | Print exp ->
+    Print (in_expression exp)
+  | Branch (exp, l1, l2) ->
+    Branch (in_expression exp, l1, l2)
+  | Osr (exp, f, v, l, osrs) ->
+    Osr (in_expression exp, f, v, l,
+         List.map in_osr osrs)
+
+  | Decl_mut (x, None)
+  | Read x ->
+    assert (x != old_name);
+    instr
+
+  | Label _ | Goto _ | Comment _ ->
+    assert (VarSet.is_empty (used_vars instr));
+    instr
+
+let freshen_assign ((_, instrs) as inp : analysis_input) (def : pc) =
+  let uses = Analysis.PcSet.elements (Analysis.used inp def) in
+  let instr = instrs.(def) in
+  match[@warning "-4"] instr with
+  | Assign (x, exp) ->
+    let fresh = fresh_var instrs x in
+    instrs.(def) <- Decl_mut (fresh, Some exp);
+    List.iter (fun pc ->
+      instrs.(pc) <- replace_uses_in_instruction x fresh instrs.(pc)) uses
+  | _ ->
+    assert(false)
+open Instr
+
+let var_in_simple_exp var (exp : simple_expression) (in_exp : simple_expression) : simple_expression =
```

Comment:
So this code was originally from `replace.ml`, and there's very similar code above from `rename.ml`. Replacing/renaming seems to be very similar, but maybe there's a reason why we can't have them share code?

But more importantly, I think these functions need to be renamed. Before, `Rename.in_simple_expression` and `Replace.var_in_simple_exp` were distinct. But now, `Edit.var_in_simple_exp` isn't clear. The names are also a little inconsistent.